Extension/lib/utils/page-stats.js

```diff
@@ -92,10 +92,192 @@ adguard.pageStats = (function (adguard) {
         pageStatsHolder.clear();
     };
 
+    /**
+     * Blocked requests types
+     */
+    var blockedTypes = {
+        OTHERS: 0,
+        ADS: 1,
+        TRACKERS: 2,
+        SOCIAL: 3,
+
+        /**
+         * Returns blocked type by filter id
+         *
+         * @param filterId
+         * @returns {number}
+         */
+        getBlockedTypeByFilterId: function (filterId) {
+            return blockedTypesFilters[filterId] ? blockedTypesFilters[filterId] : blockedTypes.OTHERS;
+        }
+    };
+
+    /**
+     * Blocked types to filters relation dictionary
+     * TODO: Fill it with correct filter ids
+     */
+    var blockedTypesFilters = {};
+    blockedTypesFilters[adguard.utils.filters.RUSSIAN_FILTER_ID] = blockedTypes.ADS;
+    blockedTypesFilters[adguard.utils.filters.ENGLISH_FILTER_ID] = blockedTypes.ADS;
+    blockedTypesFilters[adguard.utils.filters.SOCIAL_FILTER_ID] = blockedTypes.SOCIAL;
+    blockedTypesFilters[adguard.utils.filters.TRACKING_FILTER_ID] = blockedTypes.TRACKERS;
+
+    var createStatsData = function (now, type, blocked) {
+        var result = Object.create(null);
+        result.hours = [];
+        result.days = [];
+        result.months = [];
+
+        for (var i = 0; i < MAX_HOURS_HISTORY; i++) {
+            result.hours.push(createStatsDataItem(null, 0));
+        }
+        result.hours.push(createStatsDataItem(type, blocked));
+
+        for (var j = 0; j < MAX_DAYS_HISTORY; j++) {
+            result.days.push(createStatsDataItem(null, 0));
+        }
+        result.days.push(createStatsDataItem(type, blocked));
+
+        result.months.push(createStatsDataItem(type, blocked));
+
+        result.updated = now.getTime();
+
+        return result;
+    };
+
+    var createStatsDataItem = function (type, blocked) {
+        var result = new Object(null);
+        if (type) {
+            result[type] = blocked;
+        }
+        result.total = blocked;
+        return result;
+    };
+
+    var updateStatsDataItem = function (type, blocked, current) {
+        current[type] = (current[type] || 0) + blocked;
+        current.total = (current.total || 0) + blocked;
+
+        return current;
+    };
+
+    var MAX_HOURS_HISTORY = 24;
+    var MAX_DAYS_HISTORY = 30;
+
+    var updateStatsData = function (now, type, blocked, current) {
+        var currentDate = new Date(current.updated);
+
+        var result = current;
+
+        if (adguard.utils.dates.isSameHour(now, currentDate) && result.hours.length > 0) {
+            result.hours[result.hours.length - 1] = updateStatsDataItem(type, blocked, result.hours[result.hours.length - 1]);
+        } else {
+            var diffHours = adguard.utils.dates.getDifferenceInHours(now, currentDate);
```

Comment:
You can limit `diffHours` with the `MAX_DAYS_HISTORY` value, as well as `diffDays`